### ISSUE TITLE
Plans to support nn.Dropout

### ISSUE THREAD

**Issue by amin-nejad** [OpenMined/PySyft#3497]
**Is your feature request related to a problem? Please describe.**
A `syft.Plan` does not appear to properly support `nn.Dropout`. There are no errors when this layer is included in a Neural Net that inherits from `syft.Plan` but `nn.Dropout` relies on the `training` member variable from `nn.Module` to turn the dropout on and off, yet `syft.Plan` does not appear to have this data member. As far as I can tell this means dropout is always on, even when it comes to inference which is not desirable.

**Describe the solution you'd like**
Implement the `train()` and `eval()` methods to change the state of `training` like in `nn.Module`

**Describe alternatives you've considered**
None
